README.md

```diff
@@ -1 +1,28 @@
-# sig-security
\ No newline at end of file
+# SIG-Security
+
+Covers horizontal security initiatives for the OpenTelemetry project.
```

Comment:
This heavily conflicts with #1. I suggest merging that PR first and rebasing here. (Or just close that one.)